src/login/logind-action.c

```diff
@@ -135,11 +220,7 @@ int manager_handle_action(
                                        inhibit_what_to_string(m->action_what),
                                        handle_action_to_string(handle));
 
-        assert_se(target = manager_target_for_action(handle));
-
-        inhibit_operation = IN_SET(handle, HANDLE_SUSPEND, HANDLE_HIBERNATE,
-                                           HANDLE_HYBRID_SLEEP,
-                                           HANDLE_SUSPEND_THEN_HIBERNATE) ? INHIBIT_SLEEP : INHIBIT_SHUTDOWN;
+        inhibit_operation = manager_item_for_handle(handle)->inhibit_what;
```

Comment:
This assumes `manager_item_for_handle` cannot return NULL. Shouldn't `manager_item_for_handle` just have
```C
assert(handle < (ssize_t) ELEMENTSOF(action_table));
```
?

Reply:
indeed. I took that from manager_target_for_action() which weirdly had an assert for negative values, ie _HANDLE_ACTION_INVALID not passed but returned NULL for too large values.

Reply:
added assert()